src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices
+{
+    public static class MarshalTests
+    {
+        [Fact]
+        public static void AllocHGlobal_Int32_Success()
+        {
+            IntPtr p = Marshal.AllocHGlobal(100);
+            Assert.NotEqual(IntPtr.Zero, p);
+            Marshal.FreeHGlobal(p);
+        }
+
+        [Fact]
+        public static void AllocHGlobal_IntPtr_Success()
+        {
+            IntPtr p = Marshal.AllocHGlobal((IntPtr)100);
+            Assert.NotEqual(IntPtr.Zero, p);
+            Marshal.FreeHGlobal(p);
+        }
+
+        [Fact]
+        public static void AllocHGlobal_OOM()
+        {
+            if (IntPtr.Size == 8)
```

Comment:
I wanted a value that I would be sure would throw, and I couldn't be sure that int.MaxValue would.  That said, I'm probably going to remove this test, as even in its current form I think it's unreliable.